client/my-sites/domains/domain-management/list/domains-table.jsx

```diff
@@ -29,14 +29,23 @@ class DomainsTable extends PureComponent {
 		translate: PropTypes.func,
 	};
 
-	state = {
-		sortKey: 'status', // initial column to sort by - should match the header columns
-		sortOrder: -1, // initial sort order where 1 = ascending and -1 = descending
-	};
-
 	renderedQuerySiteDomains = {};
 	renderedQuerySitePurchases = {};
 
+	constructor( props ) {
+		super( props );
+
+		const queryParams = new URLSearchParams( window.location.search );
+		const queryParamSort = props.domainsTableColumns?.find(
+			( c ) => c.name === queryParams.get( 'sortKey' )
```

Comment:
Sorry, but I just realized, that if the columns come from an async request, the prop will change after the constructor runs and the query parameter will be ignored because it will be considered incorrect.

Maybe we can keep the sorting state in the hash part of the URL for simplicity?